analytics/models.py

```diff
@@ -0,0 +1,118 @@
+from django.db import models
+
+from zerver.models import Realm, UserProfile, Stream, Recipient
+from zerver.lib.str_utils import ModelReprMixin
+
+from analytics.lib.interval import MIN_TIME
+
+# would only ever make entries here by hand
+class Anomaly(ModelReprMixin, models.Model):
+    info = models.CharField(max_length=1000) # type: text_type
+
+    def __unicode__(self):
+        # type: () -> text_type
+        return u"<Anamoly: %s... %s>" % (self.info, self.id)
+
+class BaseCount(ModelReprMixin, models.Model):
+    value = models.BigIntegerField() # type: int
+    property = models.CharField(max_length=40) # type: text_type
+    end_time = models.DateTimeField() # type: datetime.datetime
+    interval = models.CharField(max_length=20) # type: text_type
+    anomaly = models.ForeignKey(Anomaly, null=True) # type: Optional[Anomaly]
+    class Meta:
+        abstract = True
+
+    @staticmethod
+    def extended_id():
+        raise NotImplementedError
+
+    @staticmethod
+    def key_model():
+        raise NotImplementedError
+
+    # future: could also have an aggregates_to function
+
+class DeploymentCount(BaseCount):
+    def extended_id():
+        return ()
+
+    def key_model():
+        return None
+
+    def __unicode__(self):
+        # type: () -> text_type
+        return u"<DeploymentCount: %s %s %s>" % (self.property, self.value, self.id)
+
+class RealmCount(BaseCount):
```

Comment:
Just noting it here so we don't forget. There should be a unique constraint on (Realm, property, end_time, interval). And unique constraint on (Stream, property, end_time, interval) on StreamCount, etc.